api/src/main/java/org/openmrs/api/context/Context.java

```diff
@@ -1389,4 +1390,49 @@ public static void setUseSystemClassLoader(boolean useSystemClassLoader) {
 	public static boolean isUseSystemClassLoader() {
 		return getServiceContext().isUseSystemClassLoader();
 	}
+
+	/**
+	 * Evicts the entity data for a particular entity "instance".
+	 *
+	 * @param sessionFactory
+	 * @since 2.6.0
+	 */
+	public static void evictSingleEntity(SessionFactory sessionFactory, Class<?> entityClass, String uuid) {
+		if(StringUtils.isBlank(uuid)) {
+			evictAllEntities(sessionFactory, entityClass);
+			return;
+		}
+		log.debug("Clearing DB cache for entity: {} with uuid: {}", entityClass, uuid);
+		sessionFactory.getCache().evictEntity(entityClass, uuid);
+		sessionFactory.getCache().evictCollectionRegions();
+		sessionFactory.getCache().evictQueryRegions();
+	}
+
+	/**
+	 * Evicts all entity data from the given region (i.e. for all entities of type).
+	 *
+	 * @param sessionFactory
+	 * @since 2.6.0
+	 */
+	public static void evictAllEntities(SessionFactory sessionFactory, Class<?> entityClass) {
+		if(entityClass == null){
+			clearEntireCache(sessionFactory);
```

Comment:
Same here, dont evict all just because enityClass is null, it defeats the method name, instead just require the class